inc/compatibility/class-compatibilityfactory.php

```diff
@@ -0,0 +1,567 @@
+<?php
+/**
+ * Compatibility Factory
+ *
+ * @package Pantheon
+ */
+
+namespace Pantheon\Compatibility;
+
+use function __;
+use function array_filter;
+use function array_key_exists;
+use function esc_html_e;
+use function get_option;
+use function in_array;
+use function time;
+use function wp_kses_post;
+use function wp_next_scheduled;
+use function wp_schedule_event;
+
+/**
+ * Class CompatibilityFactory
+ *
+ * @package Pantheon\Compatibility
+ */
+class CompatibilityFactory {
+
+
+
+	/**
+	 * Instance of Pantheon
+	 *
+	 * @var CompatibilityFactory
+	 */
+	private static $instance;
+
+	/**
+	 * @var array $plugin_classes
+	 */
+	private static $plugin_classes = [];
+
+	/**
+	 * Pantheon constructor.
+	 */
+	private function __construct() {
+		static::$plugin_classes = [
```

Comment:
I'm wondering if other things (like `$plugin_slug`) could be abstraced out of their classes into something like this, too. If we're already maintaining a list of all the fixes in the classes, we could reduce the code in those individual classes and just pass everything in. Maybe?